### PR TITLE
Add allow(unused_unsafe) to some functions.

### PR DESCRIPTION
**Pull request by vi**

### CODE REVIEW

**Comment by vi** [comment]
Or shall a bunch of `unsafe{}` blocks be removed instead? Will it cause incompatibilities for users with legacy rustc or deps?

File: src/lib.rs [hunk 1]
```diff
@@ -11,6 +11,7 @@
 #![cfg_attr(test, deny(warnings))]
 #![recursion_limit = "500"]
 #![deny(unused)]
+#![allow(unused_unsafe)]
```
Comment (RalfJung) [inline src/lib.rs:14]: Where are these unused unsafe blocks? Can't you add the lint more locally?

Comment (vi) [reply src/lib.rs:14]: I expect it to be [this](https://github.com/rust-lang/libc/pull/1870).

Shall it be fixed just for for this use case or also for possible future unsafe-to-safe conversions?

Shall it be fixed by just removing `unsafe{}` or by allowing `unused_unsafe` selectively?

Comment (asomers) [reply src/lib.rs:14]: Please do it selectively, just where we need it.  And BTW we'll have to make a 0.18.1 release just for this change.

**Review by asomers — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 1 above)
Comment (vi) [reply src/lib.rs:14]: Why a point release? Shouldn't lint caps hide those errors when nix is used as a crates.io dependency?

**Comment by vi** [comment]
Is this pull request still needed in face of #1281 which supposedly also includes similar changes?

File: src/lib.rs (hunk 1 above)
Comment (asomers) [reply src/lib.rs]: What are "lint caps"?

Comment (RalfJung) [reply src/lib.rs]: Cargo builds dependencies by calling `rustc --cap-lints allow`. This means lints are turned off when building dependencies.

**Comment by vi** [comment]
@asomers What more to be done for this?

**Review by asomers — APPROVED** [review]
LGTM!

bors r+

**Pull request merged**